catroid/src/main/java/org/catrobat/catroid/content/bricks/SetNfcTagBrick.java

```diff
@@ -0,0 +1,255 @@
+					spinnerSelection = BrickValues.TNF_WELL_KNOWN_SMS;
+				} else if (itemSelected.equals(context.getString(R.string.tnf_well_known_tel))) {
+					spinnerSelection = BrickValues.TNF_WELL_KNOWN_TEL;
+				} else if (itemSelected.equals(context.getString(R.string.tnf_well_known_mailto))) {
+					spinnerSelection = BrickValues.TNF_WELL_KNOWN_MAILTO;
+				} else if (itemSelected.equals(context.getString(R.string.tnf_external_type))) {
+					spinnerSelection = BrickValues.TNF_EXTERNAL_TYPE;
+				} else {
+					spinnerSelection = BrickValues.TNF_EMPTY;
+				}
+			}
+
+			@Override
+			public void onNothingSelected(AdapterView<?> parent) {
+			}
+		});
+
+		spinner.setSelection(spinnerSelection, true);
+		TextView textField = (TextView) view.findViewById(R.id.brick_set_nfc_tag_edit_text);
+		getFormulaWithBrickField(BrickField.NFC_NDEF_MESSAGE).setTextFieldId(R.id.brick_set_nfc_tag_edit_text);
+		getFormulaWithBrickField(BrickField.NFC_NDEF_MESSAGE).refreshTextField(view);
+		textField.setOnClickListener(this);
+
+		return view;
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		prototypeView = View.inflate(context, R.layout.brick_set_nfc_tag, null);
+
+		spinner = (Spinner) prototypeView.findViewById(R.id.brick_set_nfc_tag_ndef_record_spinner);
+		SpinnerAdapter setLookSpinnerAdapter = createArrayAdapter(context);
+		spinner.setAdapter(setLookSpinnerAdapter);
+		spinner.setSelection(spinnerSelection, true);
+
+		TextView textSetNfcTag = (TextView) prototypeView.findViewById(R.id.brick_set_nfc_tag_edit_text);
+
+		if (defaultPrototypeToken != null) {
+			int defaultValueId = InternToExternGenerator.getMappedString(defaultPrototypeToken);
+			textSetNfcTag.setText(context.getText(defaultValueId));
+		} else {
+			textSetNfcTag.setText(context.getString(R.string.brick_set_nfc_tag_default_value));
+		}
+
+		return prototypeView;
+	}
+
+	public void showFormulaEditorToEditFormula(View view) {
+		FormulaEditorFragment.showFragment(view, this, BrickField.NFC_NDEF_MESSAGE);
+	}
+
+	@Override
+	public void updateReferenceAfterMerge(Scene into, Scene from) {
+	}
+
+	private ArrayAdapter<String> createArrayAdapter(Context context) {
+		ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(context, android.R.layout.simple_spinner_item);
+
+		arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+		arrayAdapter.add(context.getString(R.string.tnf_mime_media));
+		arrayAdapter.add(context.getString(R.string.tnf_well_known_http));
+		arrayAdapter.add(context.getString(R.string.tnf_well_known_https));
+		arrayAdapter.add(context.getString(R.string.tnf_well_known_sms));
+		arrayAdapter.add(context.getString(R.string.tnf_well_known_tel));
+		arrayAdapter.add(context.getString(R.string.tnf_well_known_mailto));
+		arrayAdapter.add(context.getString(R.string.tnf_external_type));
+		arrayAdapter.add(context.getString(R.string.tnf_empty));
```

Comment:
Currently there is no consistent way to fill an array adapter nor getting the selected item in our codebase. I thought I created a Jira ticket for that but can't find it. So maybe I didn't :smile_cat:

Please take a look at:
http://stackoverflow.com/questions/4029261/populating-spinner-directly-in-the-layout-xml
or
https://github.com/Catrobat/Catroid/blob/1640aa013297914dc3c85ca5ddf009d302c70430/catroid/src/main/java/org/catrobat/catroid/physics/content/bricks/SetPhysicsObjectTypeBrick.java#L105-L107

If none of them are suited, please use [ArrayAdapter.addAll(T...)](https://developer.android.com/reference/android/widget/ArrayAdapter.html#addAll(T...))